tests/acceptance/change.spec.js

```diff
@@ -0,0 +1,319 @@
+'use strict';
+
+var expect = require('chai').expect;
+var path = require('path');
+var change = require('../../addon/ng2/utilities/change');
+var mockFs = require('mock-fs');
+var existsSync = require('exists-sync');
+var fsPromise = require('../../addon/ng2/utilities/fs-promise');
+
+describe('Change', () => {
+  var sourcePath;
+
+  describe('Invalid Path', () => {
+
+    beforeEach(() => {
+      var mockDrive = {
+        'src/app/my-component': {
+          'component-file.txt': 'hello'
+        }
+        
+      };
+      mockFs(mockDrive);
+
+      sourcePath = path.join('src/app/my-component');
+
+    });
+
+    afterEach(() => {
+      mockFs.restore();
+    });
+
+    it('the file/directory should not exist', () => {
+      var sourceFile = path.join(sourcePath, '/some_path/component-file.txt');
+      expect(existsSync(sourceFile)).to.equal(false);
+    })
+
+  });
+  
+  // Test on InsertChange
+  describe('InsertChange with valid path and valid string(s)/position to add', () => {
+
+    beforeEach(() => {
+      var mockDrive = {
+        'src/app/my-component': {
+          'component-file.txt': 'hello'
+        }
+        
+      };
+      mockFs(mockDrive);
+
+      sourcePath = path.join('src/app/my-component');
+
+    });
+
+    afterEach(() => {
+      mockFs.restore();
+    });
+
+    it('should add text to the source code and remove text from it', () => {
+      var sourceFile = path.join(sourcePath, 'component-file.txt');
+      expect(existsSync(sourceFile)).to.equal(true);
+      var changeInstance = new change.InsertChange(sourceFile, 6, ' world!');
+      return changeInstance
+        .apply()
+        .then(() => {
+          return fsPromise.readWithPromise(sourceFile);
+        }).then(contents => {
+          var expectedContents = 'hello world!';
+          expect(contents).to.equal(expectedContents);
+        }).then(()=> {
+          var anotherChangeInstance = new change.RemoveChange(sourceFile, 6, 'world!');
```

Comment:
Is there a reason these two tests are chained together? I'd like to see `InsertChange` and `RemoveChange` tested separately.